dev/com.ibm.ws.microprofile.openapi.ui/src/com/ibm/ws/microprofile/openapi/ui/OpenAPIUIEndpointManager.java

```diff
@@ -0,0 +1,192 @@
+    private static final String OPEN_API_ENDPOINT_RESERVED_PATH = "/openapi";
+
+    private static Map<String,String> CONFLICTING_PATHS_MAP = Stream.of(
+                    new AbstractMap.SimpleEntry<>("/ibm/api", "OpenAPI"),
+                    new AbstractMap.SimpleEntry<>("/health", "mpHealth"),
+                    new AbstractMap.SimpleEntry<>("/metrics","mpMetrics"))
+            .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+    
+    private static final Pattern PATH_PATTERN = Pattern.compile("^(/[\\w./_-]*)?$");
+    private static final String[] INVALID_PATHS = {"/.","/.."};
+
+    private OpenAPIWABConfigManager openAPIWABConfigManager;
+
+    @Activate
+    protected void activate(ComponentContext context, Map<String, Object> properties){
+
+        String path = OPEN_API_UI_PATH;
+        if(ProductInfo.getBetaEdition()) {
+            Config config = ConfigProvider.getConfig(OpenAPIUIEndpointManager.class.getClassLoader());
+            path = processOpenAPIUIPATH(context, config);
+        }
+        openAPIWABConfigManager = new OpenAPIWABConfigManager(context, OPEN_API_UI_VAR_Name, path);
+        openAPIWABConfigManager.pushConfiguration();
+    }
+
+    @Deactivate
+    protected void deactivate(ComponentContext context, int reason){
+        openAPIWABConfigManager.deactivate();
+    }
+
+    protected String processOpenAPIUIPATH(final ComponentContext context, final Config config) {
+        String uiPath = getOptionalValue(config, OPEN_API_UI_CONFIG_NAME, String.class, OPEN_API_UI_PATH);
+
+        uiPath = resolvePath(uiPath);
+
+        // Check if there are potential conflicts with other Liberty/MicroProfile features and warn if there is a match.
+        if (CONFLICTING_PATHS_MAP.keySet().contains(uiPath)) {
+            //just warn that might clash with other features as they might not be active, so not a cause for failure
+            Tr.warning(tc, "CWWKO1752W_OPEN_API_UI_PATH_POTENTIAL_CONFLICT", CONFLICTING_PATHS_MAP.get(uiPath));
+        }
+
+        //Validate URL for any paths that we should revert to default to
+        if(!validatePath(config, uiPath)){
+            uiPath = OPEN_API_UI_PATH;
+        }
+
+        return uiPath;
+    }
+
+    protected boolean validatePath(Config config, String path){
+        boolean valid=true;
+        //check against OpenAPI Data endpoint
+        String dataEndpoint = getOptionalValue(config, OPEN_API_ENDPOINT_CONFIG_NAME, String.class, OPEN_API_ENDPOINT_RESERVED_PATH);
+        if (path.equals(dataEndpoint)) {
+            Tr.warning(tc, "CWWKO1750W_OPEN_API_UI_PATH_RESERVED", OPEN_API_ENDPOINT_RESERVED_PATH);
+            valid=false;
+        }
+        //check against known invalid paths that contain valid characters
+        else if (Arrays.asList(INVALID_PATHS).contains(path)){
+            Tr.warning(tc, "CWWKO17510W_OPEN_API_UI_PATH_INVALID", path);
+            valid=false;
+        }
```

Comment:
This check only looks for the path being `/..` or `/.`, we should probably reject the path if it has any segment which is `..` or `.`